src/modules/npm.rs

```diff
@@ -0,0 +1,226 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::npm::NpmConfig;
+use crate::formatter::{StringFormatter, VersionFormatter};
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+use semver::Version;
+use semver::VersionReq;
+use serde_json as json;
+use std::ops::Deref;
+
+/// Creates a module with the current Node.js version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("npm");
+    let config = NpmConfig::try_load(module.config);
+    let is_npm_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .is_match();
+
```

Comment:
Modules that are disabled by default require an explicit check for `config.disabled`.